packages/core/content-manager/server/src/controllers/relations.ts

```diff
@@ -242,24 +247,58 @@ export default {
       fields: fieldsToSelect,
     };
 
+    const permissionChecker = getService('permission-checker').create({
+      userAbility,
+      model: targetedModel.uid,
+    });
+    const permissionQuery = await permissionChecker.sanitizedQuery.read(queryParams);
+
     if (isAnyToMany(attribute)) {
-      const res = await strapi.entityService.loadPages(
-        model,
-        { id },
-        targetField,
-        {
-          ...queryParams,
-          ordering: 'desc',
-        } as any,
-        {
-          page: ctx.request.query.page,
-          pageSize: ctx.request.query.pageSize,
-        }
+      const [resWithOnlyIds, res] = await Promise.all([
+        strapi.entityService.loadPages(
+          model,
+          { id },
+          targetField,
+          {
+            fields: ['id'],
+          } as any,
+          {
+            page: ctx.request.query.page,
+            pageSize: ctx.request.query.pageSize,
+          }
+        ),
+        strapi.entityService.loadPages(
+          model,
+          { id },
+          targetField,
+          {
+            ...permissionQuery,
+            ordering: 'desc',
+          } as any,
+          {
+            page: ctx.request.query.page,
+            pageSize: ctx.request.query.pageSize,
+          }
+        ),
+      ]);
```

Comment:
I believe the pagination will not properly work here  🤔 

If the permision query filters out entries that are not filtered out in the first loadPages method, pagination might be broken.

An example :
```
-- Page 1
1, 2, 3, 4, 5
1,     3, 4,   , 6, 7

-- Page 2
6, 7, 8, 9, 10
8,  9, 10, 11, 12 <--- id 7 is not available in this page, but it should
```

Reply:
indeed, I'll have to rethink that